src/ucp/api/ucp.h

```diff
@@ -251,9 +251,25 @@ enum ucp_ep_close_mode {
                                               for all endpoints created on
                                               both (local and remote) sides to
                                               avoid undefined behavior. */
-    UCP_EP_CLOSE_MODE_FLUSH         = 1  /**< @ref ucp_ep_close_nb schedules
+    UCP_EP_CLOSE_MODE_FLUSH         = 1, /**< @ref ucp_ep_close_nb schedules
                                               flushes on all outstanding
                                               operations. */
+    UCP_EP_CLOSE_MODE_STREAM_CLOSE   = 2 /**< @ref ucp_ep_close_nb will close
+                                              the stream connection. The remote
+                                              side will have a @ref
+                                              UCP_STREAM_POLL_FLAG_CLOSED flag
+                                              for the remote endpoint if @ref
+                                              ucp_stream_worker_poll is called
+                                              on the remote worker. This event
+                                              puts the remote endpoint into an
+                                              error state, all following @ref
+                                              ucp_stream_send_nb calls on the
+                                              remote endpoint will return @ref
+                                              UCS_ERR_REMOTE_DISCONNECT error
+                                              code. The UCP should be
+                                              initialized with @ref
+                                              UCP_FEATURE_STREAM, otherwise
+                                              using of stream API is invalid. */
```

Comment:
I suggest the following re-wording for clarity:
`@ref` ucp_ep_close_nb will close the stream connection. The remote side will return `@ref` UCP_STREAM_POLL_FLAG_CLOSED for the remote endpoint if `@ref` ucp_stream_worker_poll is called on the remote worker. This mode puts the remote endpoint into an error state. All subsequent `@ref` ucp_stream_send_nb calls on the remote endpoint will return `@ref` UCS_ERR_REMOTE_DISCONNECT. 
`@note` To use the stream API, UCP should be initialized with `@ref` UCP_FEATURE_STREAM, otherwise the use of the stream API is invalid.